pkg/executables/kubectl.go

```diff
@@ -994,3 +994,21 @@ func (k *Kubectl) setImage(ctx context.Context, kind, name, container, image str
 
 	return nil
 }
+
+func (k *Kubectl) CheckProviderExists(ctx context.Context, kubeconfigFile, name, namespace string) (bool, error) {
```

Comment:
Unit test/s for this method? I'm ok adding them in a separate PR

Reply:
will do in a separate PR for this and the new method in installer.go